c/json.c

```diff
@@ -1760,10 +2154,81 @@ int jsonAsBoolean(Json *json) {
 int jsonAsNumber(Json *json) {
   if (json->type == JSON_TYPE_NUMBER) {
     return json->data.number;
+  } else if (json->type == JSON_TYPE_INT64){
+    return json->data.integerValue;
   }
   return 0;
 }
 
+int64_t jsonAsInt64(Json *json){
+  switch (json->type){
+  case JSON_TYPE_NUMBER:
+    return (int64)json->data.number;
+  case JSON_TYPE_INT64:
+    return json->data.integerValue;
+  case JSON_TYPE_DOUBLE:
+    return (int64)json->data.floatValue;
+  default:
+    return 0;
+  }
+}
+
+double jsonAsDouble(Json *json){
+  switch (json->type){
+  case JSON_TYPE_NUMBER:
+    return (double)json->data.number;
+  case JSON_TYPE_INT64:
+    return (double)json->data.integerValue;
+  case JSON_TYPE_DOUBLE:
+    return json->data.floatValue;
+  default:
+    return 0;
+  }
+}
+
+static int64 hashString(char *s){
+  int64 hash = 5381;
+  int i;
+  int len = strlen(s);
```

Comment:
Any chance the string is huge and we'd want to use `size_t`? I don't think there is any penalty, specially considering `strlen` returns `size_t`.

Reply:
I hope not.  A key with size > 1 GB.  What is the world coming to?

Reply:
Ok, I do think we should be just using `size_t` for sizes like the C library, maybe someone will change it in the future.